packages/secure_storage/amplify_secure_storage/lib/src/amplify_secure_storage.vm.dart

```diff
@@ -27,25 +33,66 @@ class AmplifySecureStorage extends AmplifySecureStorageInterface {
     required super.config,
   });
 
-  late final SecureStorageInterface _instance = () {
-    if (Platform.isAndroid) {
-      return AmplifySecureStorageAndroid(config: config);
-    }
-    return AmplifySecureStorageWorker(config: config);
-  }();
+  late final AmplifySecureStorageInterface _instance;
+
+  final _initMemo = AsyncMemoizer();
+
+  Future<void> _init() async {
+    await _initMemo.runOnce(
+      () async {
+        if (Platform.isAndroid) {
+          _instance = AmplifySecureStorageAndroid(config: config);
+        } else {
+          final applicationDirectory = await getApplicationSupportDirectory();
+          _instance = AmplifySecureStorageWorker(
+            config: config,
+            applicationDirectory: applicationDirectory.path,
+          );
+        }
+        if (Platform.isLinux) {
+          await _initializeScope(config.linuxOptions.accessGroup);
+        }
+      },
+    );
+  }
 
   @override
-  FutureOr<void> delete({required String key}) {
+  Future<void> delete({required String key}) async {
+    await _init();
     return _instance.delete(key: key);
   }
 
   @override
-  FutureOr<String?> read({required String key}) {
+  Future<String?> read({required String key}) async {
+    await _init();
     return _instance.read(key: key);
   }
 
   @override
-  FutureOr<void> write({required String key, required String value}) {
+  Future<void> write({required String key, required String value}) async {
+    await _init();
     return _instance.write(key: key, value: value);
   }
+
+  /// Clears all keys for the given scope if this scope
+  /// has not been initialized previously.
+  ///
+  /// Checks for an initialization flag in file storage.
+  /// If the flag is not present storage will be cleared
+  /// and then the flag will be set.
+  ///
+  /// Intended to clear storage after an app uninstall & re-install.
+  Future<void> _initializeScope(String? accessGroup) async {
+    // if accessGroup is set, do not clear data on initialization
+    // since the data can be shared across applications.
+    if (accessGroup != null) return;
```

Comment:
Can we reformat this to not call `_initalizeScope` if accessGroup is set since the only reason this method has a parameter is to check if it's not null which is kind of confusing.